x-pack/plugins/reporting/server/config/schema.test.ts

```diff
@@ -112,4 +112,10 @@ describe('Reporting Config Schema', () => {
       ConfigSchema.validate({ export_types: { csv: { enabled: true } } }, { dev: true })
     ).not.toThrow();
   });
+
+  it('should have roles enabled set to false for serverless by default', () => {
+    expect(() =>
+      ConfigSchema.validate({ roles: { enabled: false } }, { serverless: true })
+    ).not.toThrow();
+  });
```

Comment:
This test should pass an empty object as the first param. The second param should be as you have it. Running validation on the empty object will return a config object with all of the defaults. We want to look at the defaults that are returned, and make sure they are the correct setting for the environment.

```suggestion
  describe('roles', () => {
    it('should have roles enabled set to false for serverless by default', () => {
      expect(ConfigSchema.validate({}, { serverless: true }).roles.enabled).toBe(false);
    });

    it('should have roles enabled set to true for non-serverless by default', () => {
      expect(ConfigSchema.validate({}).roles.enabled).toBe(true);
    });
  });
```